dhis-2/dhis-support/dhis-support-commons/src/test/java/org/hisp/dhis/commons/config/jackson/JacksonObjectMapperConfigTest.java

```diff
@@ -81,6 +82,18 @@ public void testNullDate()
         assertNull( yearTest.get( "date" ) );
     }
 
+    @Test // DHIS2-8582
+    public void testSerializerUserWithUser()
+        throws JsonProcessingException
+    {
+        User user = new User();
+        user.setAutoFields();
+        user.setUser( user );
+        user.setLastUpdatedBy( user );
+
+        jsonMapper.writeValueAsString( user );
```

Comment:
Is this test complted? Looks like you are not asserting anything

Reply:
It would normally throw an exception if the circular ref was still an issue, not really much to assert here

Reply:
Made some fixes to it, should be ok (also asserts that it can be deserialized)